### DISCUSSION TITLE
Wix Toolset version 5 with Managed BootstrapperApplication (created in .NET Framework 4.8) mbanative payload issue

### DISCUSSION THREAD

**Discussion by harshmodhiya** [wixtoolset/issues#8727] [Questions]
We have a Managed Bootstrapper Application with the below version:

- MBA - .NET Framework 4.8.1 (Platform Any CPU)
- WiX toolset - 5 (Platform x86)

**Here's the scenario in our application:**
I want to add 2 mbanative.dll payloads as per the platform:
`<Payload SourceFile="$(var.UI.TargetDir)runtimes\win-x64\native\mbanative.dll" />
<Payload SourceFile="$(var.UI.TargetDir)runtimes\win-x86\native\mbanative.dll" />`

But I can't because of the below error:
_Error (active)	WIX8002	The Payload 'pay7TUh8xmtuMGYxpvZ2DTsOj3qwks' has a duplicate Name 'mbanative.dll' in the BA container. When extracting the container at runtime, the file will get overwritten._

If I use $(var.Platform), it takes 1 payload in the executable file at a time. So, it will not work with different platforms.
**Need help with the below,**

- How can we manage both platform's DLL in 1 executable?

**Comment by robmen** [comment]
Why?

**Reply by harshmodhiya** [reply]
Reason behind this is "When an executable runs on a 32-bit PC, it requires a 32-bit mbnative dll; on a 64-bit PC, it requires a 64-bit one."

**Reply by robmen** [reply]
> When an executable runs on a 32-bit PC, it requires a 32-bit mbnative dll; on a 64-bit PC, it requires a 64-bit one.

That's not how DLLs work. DLLs need to match the architecture of the executable that loads them.
